Add silent option

MadeByMike opened the pull request:
@necolas sorry about deleting the other pull request. Added test and rebased on master.
Re: https://github.com/reworkcss/css-parse/issues/70

MadeByMike committed:
add silent option

MadeByMike committed:
better ignore rules that fail

MadeByMike committed:
delete tmp file

MadeByMike committed:
more better ignore errors

necolas commented on index.js:
```diff
@@ -69,6 +69,9 @@ module.exports = function(css, options){
    */
 
   function error(msg) {
+    if(options.silent === true){
```
space after if. space before {

necolas commented on index.js:
```diff
@@ -69,6 +69,9 @@ module.exports = function(css, options){
+    if(options.silent === true){
+      return false;
+    }
     var err = new Error(msg + ' near line ' + lineno + ':' + column);
```
insert new line

necolas commented on index.js:
```diff
@@ -116,8 +119,10 @@ module.exports = function(css, options){
     while (css.length && css.charAt(0) != '}' && (node = atrule() || rule())) {
-      rules.push(node);
-      comments(rules);
+      if(node !== false){
```
spaces either side of parens

necolas commented on index.js:
```diff
@@ -150,7 +155,11 @@ module.exports = function(css, options){
     var c;
     rules = rules || [];
-    while (c = comment()) rules.push(c);
+    while (c = comment()){
```
space before {

necolas commented on index.js:
```diff
@@ -245,8 +254,10 @@ module.exports = function(css, options){
     while (decl = declaration()) {
-      decls.push(decl);
-      comments(decls);
+      if(decl !== false){
```
whitespace again

necolas commented on index.js:
```diff
@@ -245,8 +254,10 @@ module.exports = function(css, options){
+      if(decl !== false){
+        decls.push(decl);
+        comments(decls);
+	    }
```
indentation

necolas commented on test/css-parse.js:
```diff
@@ -66,4 +66,13 @@ describe('parse(str)', function(){
+      parse('thing { color: red; } /* b { color: blue; }',{ silent: true });
+    });
+  });
+
```
remove the new lines

MadeByMike committed:
code formatting

my opinionated IDE and lack of attention to detail have upset Nicolas Gallagher and that upsets me :). I hope I found them all!

MadeByMike committed:
code formatting

necolas commented:
Applied in 72f4293401ade3011c46b6a23007fb451e1cad04
